src/main/kotlin/se/uulm/snowballr/backend/validation/ProjectPaperValidator.kt

```diff
@@ -0,0 +1,43 @@
+package se.uulm.snowballr.backend.validation
+
+import arrow.core.EitherNel
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+import arrow.core.raise.zipOrAccumulate
+import se.uulm.snowballr.backend.model.ValidationIssue
+import snowballr.ProjectOuterClass.Project.Paper as GrpcProjectPaper
+
+/**
+ * A validator for [GrpcProjectPaper] related requests.
+ */
+object ProjectPaperValidator {
+    fun validateGetRequest(request: GrpcProjectPaper.Get): EitherNel<ValidationIssue, Unit> = either {
+        zipOrAccumulate(
+            { ensureIdValidity("id", request.projectId) },
```

Comment:
This is the incorrect field name.
```suggestion
            { ensureIdValidity("project_id", request.projectId) },
```

Reply:
Oh shit. Good catch.